tests/corpus/parsing/test_cha.py

```diff
@@ -1,46 +1,38 @@
-import os
-import tempfile
 import pytest
-import requests
 from sktalk.corpus.conversation import Conversation
 from sktalk.corpus.parsing.cha import ChaFile
 
 
-class TestChaFile:
-    urls = [
-        "https://ca.talkbank.org/data-orig/GCSAusE/01.cha",
-        "https://ca.talkbank.org/data-orig/GCSAusE/02.cha",
-        "https://ca.talkbank.org/data-orig/GCSAusE/03.cha"
-    ]
-
-    @pytest.fixture(params=urls)
-    def download_file(self, request):
-        remote = request.param
-        response = requests.get(remote, timeout=5)
-        response.raise_for_status()
+@pytest.fixture
+def path_source():
+    return "tests/testdata/file01.cha"
 
-        ext = os.path.splitext(remote)[1]
 
-        with tempfile.NamedTemporaryFile(delete=False, suffix=ext) as temp_file:
-            temp_file.write(response.content)
-            temp_file_path = temp_file.name
+@pytest.fixture
+def cha_info():
+    n_utterances = 13
+    participants = {'A', 'B'}
+    languages = ['eng']
+    return n_utterances, participants, languages
 
-        yield temp_file_path
 
-        os.remove(temp_file_path)
-
-    @pytest.mark.parametrize("download_file", urls, indirect=True)
-    def test_parse(self, download_file):
-        parsed_cha = ChaFile(download_file).parse()
+class TestChaFile:
+    def test_parse(self, path_source, cha_info):
+        n_utterances, participants, languages = cha_info
+        cha_utts, cha_meta = ChaFile(path_source).parse()
+        assert cha_meta["source"] == path_source
+        assert {u.participant for u in cha_utts} == participants
+        assert cha_meta["Languages"] == languages
+        assert len(cha_utts) == n_utterances
+
+    def test_wrapped_parser(self, path_source, cha_info):
+        n_utterances, participants, languages = cha_info
+        parsed_cha = Conversation.from_cha(path_source)
         assert isinstance(parsed_cha, Conversation)
-        source = parsed_cha.metadata["source"]
-        assert os.path.splitext(source)[1] == ".cha"
-        assert parsed_cha.utterances[0].begin == "00:00:00.000"
-        participant = parsed_cha.utterances[0].participant
-        assert participant in ["A", "B", "S"]
-        language = parsed_cha.metadata["Languages"]
-        assert language == ["eng"]
-        # TODO assert that there are no empty utterances
+        assert parsed_cha.metadata["source"] == path_source
+        assert {u.participant for u in parsed_cha.utterances} == participants
+        assert parsed_cha.metadata["Languages"] == languages
+        assert len(parsed_cha.utterances) == n_utterances
```

Comment:
As above, I would replace the sequence of specific assertions with a single assertion per resulting object to prevent any properties changing unexpectedly.